datamade/ccfp-asset-dashboard
# Added a project category, phase completion, districts, and zones to the Project form

## Overview
This PR adds the project category, phase completion, districts, and zones to the project form. It's based on the list in #21. There's a little bit of styling for the PR.

### Demo
![project-districts](https://user-images.githubusercontent.com/38969506/105181840-ed501a00-5af1-11eb-8674-52e618986424.gif)


### Notes
I wanted to get this PR ok'd to make sure that I changed all of the district and project models correctly. I also need a check on how I'm doing the forms. 

The list from #21 has "Need" and "Score Weight" but I'm not sure how those fit into this form.

There are some tasks that I need to do to complete this PR and close out #21:
1. Test the formset and validation. The review app won't deploy with broken tests.
2. Add custom validation to the model fields (a regex to parse the input so that it matches "District 15" etc.)
3. Finish the javascript where a user can add an additional field to the districts.
4. Fix the bug where a user can clear out a district's name and save the form, which prompts an error. The error message renders on the screen but the deleted text is still missing. The fix for this is related to number 3 ^.

I had some issues making the project category fields with dynamic selections for the subcategory, so I made it show each category + subcategory within a drop-down selector. Those current options came from the raw CIP csv.

## Testing Instructions
The current code won't deploy a review app because the tests are failing on the new formsets.
- Pull the changes to your local computer
- Build the containers
- Load the fixtures: `docker-compose run --rm app python manage.py load_development_data` 
- Visit localhost:8000/projects/
- Add a project
- Update the project within the form
- Make sure that all of the parts of the form work, and not just the new fields.
